src/main/io/osd.c

```diff
@@ -1401,6 +1423,13 @@ STATIC_UNIT_TESTED void osdRefresh(timeUs_t currentTimeUs)
     static timeUs_t osdStatsRefreshTimeUs;
     static uint16_t endBatteryVoltage;
 
+#ifdef USE_DSHOT
+    if (isTryingToArm() && !ARMING_FLAG(ARMED)) {
+        osdShowTryingToArm(currentTimeUs);
```

Comment:
This will suppress all other information for the duration of the delay?
I don't think it is helpful to withhold this information from pilots for this time. Should we rather add this as a warning?

Reply:
Yes, the screen will blank except for the message and countdown.  Note that during this time we're not armed so there's no useful flight information.  The idea is to give immediate feedback to the arming switch so the pilot doesn't think arming failed.  So it's meant to be "in your face" that it will be arming momentarily.  Changing it to use a warning is risky because the user could choose to not have warnings displayed.

The secondary value is provide indirect pressure on the pilots to change their configuration to avoid having the beacon active at arming.  By improving their configuration they can avoid this delay that they see every time.

Reply:
Not sure about the value of being so prescriptive and unflexible for the 'ARMED' warning. The same effect could be achieved by making it configurable and defaulting it to on - the pilot could disable it, but that would have to be a conscious decision. Plus they could move it to wherever they like.

But I agree that this is something for after 3.4.

Re the Dshot beeper warning, I don't see why this should be so much more prominent than the other warnings, including all of the other arming disabled reasons.

Reply:
Well I did actually initially think about making it a warning.  But because of:
1. User could disable warnings
2. Warning size is only 11 characters
3. Not really a "arming disabled" but "arming delayed" message

Item 1 was the big issue.  For item 2 I would need to increase a few bytes to maintain the "ARMING In x.y" countdown feedback.  But this wouldn't provide the opportunity to display why the delay was happening like I do now with showing "DISABLING BEACON".  Item 3 is just semantics.

So item 1 was the biggest deal for me, followed by the loss in information for item 2.  But I'm open to changing it to a warning if we want to accept these issues.  I think it would need to override other possible warnings except for arming disabled reasons - but that will automatically happen becasue `isTryingToArm()` will never be true in these circumstances.

Also as you may have seen I opened an issue and called out Felix and Stefan to see if we can't fix this problem in the ESC firmware going forward.  Hopefully that'll be our long-term solution and we can eliminate this arming delayed code completely.

Reply:
@etracer65:

Re 1.: The same applies to other warnings - but they are on by default, and the expectation is that if the user wilfully disables them they are aware that they won't get them. Plus there is also an up side to this - if we add it to the warnings list that can be disabled, we get another place in the configurator where the users can see the warning, and read an eventual tooltip, so they know what's happening before it's happening (and they will have to go there in order to disable the warning).

Re 2.: 'BEACON ON' and 'ARM IN x.y' will fit, without much loss of information. And the warning list is in order of priority, so we can make sure it gets seen.

Reply:
did you consider animating ARMED message?
It will be minimal change to UI and user will be notified that arming is pending for some reason ...

Reply:
@mikeller Ok, I will look into changing into a warning.  The warning field is only defined to be 11 characters so I'm not sure how to display both messages - "BEACON ON" and the "ARM IN x.y" countdown.  I suppose I could display the "BEACON ON" on one row above but there's a slight risk the pilot has the warnings positioned at the top of the screen.

Reply:
Ah, I meant to first flash 'BEACON ON', then count down.